### PR TITLE
Fix dependency conflicts

### PR DESCRIPTION
**Pull request by ssbarnea**
This backports changes already made to master via: #1549 and #1548

### CODE REVIEW

**Comment by SharpEdgeMarshall** [comment]
We should backport CI updates too

**Comment by ssbarnea** [comment]
> We should backport CI updates too

Ohh 💩...., that may require an amount of work that nobody will be able to do. The hidden cost of not making any release in more than an year.

@simobasso What can we do to unblock this? The jinja2 and MarkupSafe new releases are causing lots of failures for cookiecutter users. I already found CI/CD on two projects I use and another user made a comment on my fix aking about the backport.

I am wondering if it would not make sense to perform this hotfix manually, including releasing it. It should buy us enough time to make the 2.0.0 release as I doubt we will see many similar problems very soon. It work nothing that these were the only dependencies that were pinned down.

**Comment by SharpEdgeMarshall** [comment]
We could try to backport CI updates, it should not be a big deal.

**Comment by ssbarnea** [comment]
@SharpEdgeMarshall Apparently GHA refuses to start running any workflows on this branch, even after I updated the conditions. It is supposed to run old jobs names like linting, ubuntu-py27,... but it triggers nothing, the only one listed below are those that I marked as required in branch rules but if you look at actions tab, you will not find anything triggered.

**Comment by ssbarnea** [comment]
After manually adding the branch config I was able to trigger the workflow but only on push (after the merge) see https://github.com/cookiecutter/cookiecutter/actions/runs/841530404 

That PR made from my fork towards this the stable branch still does not trigger a thing. I wonder what else I am missing.

**Comment by ssbarnea** [comment]
Found! Apparently putting `pull_request: {}` works. Somehow branches: * bit was preventing it from triggering the workflows... another badly documented area on the workflows.

**Comment by SharpEdgeMarshall** [comment]
Actually is documented [here](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags):
>
'*' | Matches all branch and tag names that don't contain a slash (/). The * character is a special character in YAML. When you start a pattern with *, you must use quotes. | mainreleases
-- | -- | --
'**' | Matches all branch and tag names. This is the default behavior when you don't use a branches or tags filter. | all/the/branchesevery/tag

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,8 +8,6 @@ on:
     tags:
       - "*"
   pull_request:
-    branches:
```
Comment (SharpEdgeMarshall) [inline .github/workflows/main.yml:11]: Keep the old version, just try with `**`

**Comment by ssbarnea** [comment]
@SharpEdgeMarshall Do not forget to approve the PR, so I can merge it w/o using admin rights.

**Pull request merged**